src/js/structs/Transaction.js

```diff
@@ -1,5 +1,12 @@
+import FormActionTypes from '../constants/TransactionTypes';
+
+const validTypes = Object.values(FormActionTypes);
+
 class Transaction {
-  constructor(path, value, type = 'SET') {
+  constructor(path, value, type = FormActionTypes.SET) {
+    if (!validTypes.includes(type)) {
+      throw new TypeError(`Only the following types are allowed: ${validTypes}`);
```

Comment:
The error message right now is:
```
TypeError: Only the following types are allowed: Symbol(SET),Symbol(DEL)
```
I am okay with changing it to `SET, DEL, ...` , if that is still clear enough that only the predefined values are valid types and not strings.